### PR TITLE
feat(okta): list all users

### PR DESCRIPTION
**Pull request by dannylwe**
## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is:
- [ ] External API requests have `retries`
- [ ] Pagination is used where appropriate
- [ ] The built in `nango.paginate` call is used instead of a `while (true)` loop
- [ ] Third party requests are NOT parallelized (this can cause issues with rate limits)
- [ ] If a sync requires metadata the `nango.yaml` has `auto_start: false`
- [ ] If the sync is a `full` sync then `track_deletes: true` is set

### CODE REVIEW

File: integrations/okta/nango.yaml [hunk 1]
```diff
@@ -29,6 +29,12 @@ integrations:
                 input: OktaAssignRemoveUserGroup
                 scopes:
                     - okta.groups.manage
+            list-all-users:
```
Comment (khaliqgant) [inline integrations/okta/nango.yaml:32]: just call this `users`

File: integrations/okta/nango.yaml [hunk 2]
```diff
@@ -29,6 +29,12 @@ integrations:
...
+                description: Lists users in your org
+                output: OktaListUsersResponse
+                endpoint: GET /users
+                scopes:
+                    - okta.groups.manage
```
Comment (khaliqgant) [inline integrations/okta/nango.yaml:37]: `okta.users.read`

File: integrations/okta/actions/list-all-users.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import type { NangoAction, ProxyConfiguration } from '../../models';
+import { OktaListUsersResponse, OktaUser } from '../types';
+
+export default async function runAction(nango: NangoAction): Promise<OktaListUsersResponse> {
+    const config: ProxyConfiguration = {
+        // https://developer.okta.com/docs/api/openapi/okta-management/management/tag/User/#tag/User/operation/listUsers
+        endpoint: `/api/v1/users`,
```
Comment (khaliqgant) [inline integrations/okta/actions/list-all-users.ts:7]: Add in pagination and also filter on `lastUpdated` property to make it an incremental sync https://developer.okta.com/docs/api/openapi/okta-management/management/tag/User/#tag/User/operation/listUsers!in=query&path=filter&t=request

File: integrations/okta/nango.yaml [hunk 4]
```diff
@@ -29,6 +29,12 @@ integrations:
                 input: OktaAssignRemoveUserGroup
                 scopes:
                     - okta.groups.manage
+            list-all-users:
+                description: Lists users in your org
+                output: OktaListUsersResponse
```
Comment (khaliqgant) [inline integrations/okta/nango.yaml:34]: can just be

`output: User`

models:
   User:
      id: string
      ...

File: integrations/okta/nango.yaml (hunk 1 above)
Comment (dannylwe) [reply integrations/okta/nango.yaml]: resolved

File: integrations/okta/nango.yaml (hunk 2 above)
Comment (dannylwe) [reply integrations/okta/nango.yaml]: resolved

File: integrations/okta/actions/list-all-users.ts (hunk 3 above)
Comment (dannylwe) [reply integrations/okta/actions/list-all-users.ts]: done

File: integrations/okta/nango.yaml (hunk 4 above)
Comment (dannylwe) [reply integrations/okta/nango.yaml]: done

File: integrations/okta/nango.yaml [hunk 5]
```diff
@@ -83,3 +92,5 @@ models:
             secondEmail: string | null
             login: string
             email: string
+    OktaListUsersResponse:
```
Comment (khaliqgant) [inline integrations/okta/nango.yaml:95]: No longer needed

**Comment by dannylwe** [comment]
handled here https://github.com/NangoHQ/integration-templates/pull/98, closing because of incorrect branch name

**Pull request closed without merging**